test/test_rulesets.jl

```diff
@@ -43,14 +43,30 @@ config = YotaRuleConfig()
     end
 
     @testset "getfield & co." begin
-        p = PointRS(42.0, 54.0)
+        p = PointRS(42.0+im, 54.0) # different types to test inference
         test_rrule(config, getproperty, p, :x; check_inferred=false)
         test_rrule(config, getfield, p, :x; check_inferred=false)
 
         t = (2.0, 4.0, 8.0)
         test_rrule(config, getfield, t, 3; check_inferred=false)
 
-        test_rrule(config, __new__, PointRS, 42.0, 54.0; check_inferred=false)
+        PT = PointRS{ComplexF64, Float64}
+        test_rrule(config, __new__, PT, 42.0+im, 54.0; check_inferred=false)
+
+        # These tests must be done separately because test_rrule could break constprop
+        if VERSION > v"1.6"
```

Comment:
Should it be `VERSION > v"1.6"`? I see this block [fails](https://github.com/dfdx/Yota.jl/actions/runs/4209683436/jobs/7313170779#step:5:190) in Julia 1.6.

Reply:
I wonder if `@testset` is hoisting the tests outside of the conditional somehow. Will look into this.